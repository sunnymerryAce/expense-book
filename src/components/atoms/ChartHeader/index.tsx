import React from "react";
import Box from "@material-ui/core/Box";
import getPriceSeparatedByComma from "@/util/functions/getPriceSeparatedByComma";
import useStyles from "./style";

export interface ChartHeaderProps {
  expenseAmount: number;
  budgetAmount: number;
}

const ChartHeader: React.FC<ChartHeaderProps> = (props) => {
  const classes = useStyles();
  return (
    <Box className={classes.root} fontFamily="h2.fontFamily">
      <Box>Total: ¥{getPriceSeparatedByComma(props.expenseAmount)}</Box>
      <Box padding="0 10px">/</Box>
      <Box>Budget: ¥{getPriceSeparatedByComma(props.budgetAmount)}</Box>
    </Box>
  );
};

export default ChartHeader;
