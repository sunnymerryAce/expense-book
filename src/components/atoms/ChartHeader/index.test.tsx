import React from "react";
import ChartHeader from "./";
import { shallow } from "enzyme";

describe("<ChartHeader />", () => {
  let container: any = null;
  const props = {
    expenseAmount: 200000,
    budgetAmount: 300000,
  };

  beforeEach(() => {
    container = shallow(<ChartHeader {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should render total amount", () => {
    expect(container.find(".ChartHeader-total").text()).toBe("Total: ¥200,000");
  });

  test("should render budget amount", () => {
    expect(container.find(".ChartHeader-budget").text()).toBe(
      "Budget: ¥300,000"
    );
  });
});
