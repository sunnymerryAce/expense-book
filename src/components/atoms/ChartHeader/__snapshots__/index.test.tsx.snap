// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartHeader /> should match the snapshot 1`] = `"<div class=\\"MuiBox-root MuiBox-root-3 makeStyles-root-1\\"><div class=\\"MuiBox-root MuiBox-root-4 ChartHeader-total\\">Total: ¥200,000</div><div class=\\"MuiBox-root MuiBox-root-5\\">/</div><div class=\\"MuiBox-root MuiBox-root-6 ChartHeader-budget\\">Budget: ¥300,000</div></div>"`;
